src/coreclr/jit/morph.cpp

```diff
@@ -10887,28 +10894,47 @@ GenTree* Compiler::fgOptimizeHWIntrinsic(GenTreeHWIntrinsic* node)
         }
     }
 
-    // Transforms:
-    // 1.(~v1 & v2) to VectorXxx.AndNot(v1, v2)
-    // 2.(v1 & (~v2)) to VectorXxx.AndNot(v2, v1)
-    switch (node->HWOperGet())
+    bool       isScalar = false;
+    genTreeOps oper     = node->HWOperGet(&isScalar);
+
+    if (isScalar)
     {
+        return node;
+    }
+
+    switch (oper)
```

Comment:
There's many other transforms we could do as well, but I just wanted to cover the core ones that GenTreeOp already had to start